panther_power_control/src/power_board_node.py

```diff
@@ -52,26 +33,56 @@ def turn_on(self) -> None:
     def turn_off(self) -> None:
         self._enabled = False
         self._last_state = False
-        GPIO.output(self._pin, self._last_state)
+        self._watchdog_pin.set_value(self._last_state)
 
 
 class PowerBoardNode:
     def __init__(self, name: str) -> None:
-        rospy.init_node(name, anonymous=False)
+        self._node_name = name
+        rospy.init_node(self._node_name, anonymous=False)
 
         self._pins_lock = Lock()
         self._e_stop_lock = Lock()
 
         self._clearing_e_stop = False
-        self._pins = PatherGPIO()
 
-        self._setup_gpio()
+        out_line_names = [
+            'AUX_PW_EN',  # Enable auxiliary power, eg. supply to robotic arms etc.
+            'CHRG_DISABLE',  # Disable charger
+            'DRIVER_EN',  # Enable motor drivers (1 - on)
+            'FAN_SW',  # Turn on the fan (1 - on)
+            'VDIG_OFF',  # Turn the digital power off eg. NUC, Router etc. (1 - off)
+            'VMOT_ON',  # Enable mamin power supply to motors (1 - on)
+            'WATCHDOG',  # Watchdog pin, if PWM is on this pin Panther will work
+        ]
+        in_line_names = [
+            'CHRG_SENSE',  # Charger sensor (1 - charger plugged in)
+            'E_STOP_RESET',  # Works as IN/OUT,
+            # IN - gives info if E-stop is on (1 - off), OUT - send 1 to reset estop
+            'SHDN_INIT',  # Shutdown Init managed by systemd service
+        ]
+
+        self._chip = gpiod.Chip('gpiochip0', gpiod.Chip.OPEN_BY_NAME)
+        self._lines = {name: self._chip.find_line(name) for name in out_line_names + in_line_names}
+        for name, line in self._lines.items():
+            line.request(
+                self._node_name,
+                gpiod.LINE_REQ_DIR_OUT if name in out_line_names else gpiod.LINE_REQ_DIR_IN,
+            )
+
+        self._lines['CHRG_DISABLE'].set_value(True)
+        self._lines['AUX_PW_EN'].set_value(False)
+        self._lines['FAN_SW'].set_value(False)
+        # This pin is inverted so default has to be high
+        self._lines['VDIG_OFF'].set_value(False)
+
+        self._watchdog = Watchdog(self._lines['WATCHDOG'])
         self._motor_start_sequence()
-        self._watchdog = Watchdog(self._pins.WATCHDOG)
 
         self._gpio_wait = 0.05  # seconds
+        self._last_e_stop_state = not self._lines['E_STOP_RESET'].get_value()
         self._e_stop_interrupt_time = float('inf')
-        self._chrg_sense_interrupt_time = float('inf')
+        self._chrg_sense_pressed_time = float('inf')
```

Comment:
This relates to shutdown button, shouldn't it be sth like `shdn_init_pressed_time`